ansible/roles/test/files/ptftests/mtu_test.py

```diff
@@ -0,0 +1,148 @@
+'''
+Description:    This file contains the MTU test for SONIC
+
+Usage:          Examples of how to use log analyzer
+                ptf --test-dir ptftests mtu_test.MtuTest  --platform remote  -t "testbed_type='t1-lag';" --relax --debug info --log-file /tmp/mtu_test.log  --disable-vxlan --disable-geneve --disable-erspan --disable-mpls --disable-nvgre --socket-recv-size 16384
+
+'''
+
+#---------------------------------------------------------------------
+# Global imports
+#---------------------------------------------------------------------
+import logging
+import ptf
+import ptf.packet as scapy
+import ptf.dataplane as dataplane
+
+from ptf import config
+from ptf.base_tests import BaseTest
+from ptf.mask import Mask
+from ptf.testutils import *
+
+
+class MtuTest(BaseTest):
+    '''
+    @summary: Overview of functionality
+    Test sends a jumbo ICMP echo frame with MAX MTU size and expects the reply
+    back. It also sends a jumbo frame to a route destination for verifying the 
+    forwarding functionality
+    
+    For the device configured with IP-MTU=9100, PHY-MTU=9122, 
+     - ICMP frame, the packet-len is 9114 (Subtracting 8 bytes of ICMP header from 9122)
+     - IP frame, the packet-len is 9122 (This includes the Layer 2 Ethernet header + FCS)
+    '''
+
+    #---------------------------------------------------------------------
+    # Class variables
+    #---------------------------------------------------------------------
+    DEFAULT_PACKET_LEN = 9122
+    ICMP_HDR_LEN = 8
+
+    def __init__(self):
+        '''
+        @summary: constructor
+        '''
+        BaseTest.__init__(self)
+        self.test_params = test_params_get()
+
+    #---------------------------------------------------------------------
+
+    def setUp(self):
+        self.dataplane = ptf.dataplane_instance
+        self.router_mac = self.test_params['router_mac']
+    
+    #---------------------------------------------------------------------
+
+    def check_icmp_mtu(self):
+        '''
+        @summary: Check ICMP/Ping works for MAX MTU.
+        '''
+        ip_src = "10.0.0.1"
+        ip_dst = "10.0.0.0"
+        src_mac = self.dataplane.get_mac(0, 0)
+
+        pktlen = (self.DEFAULT_PACKET_LEN - self.ICMP_HDR_LEN)
+
+        pkt = simple_icmp_packet(pktlen=pktlen,
```

Comment:
should we set the DF bit in the packet to make sure we are not fragment the packet anywhere and we were not really testing the MTU sized packet?

Reply:
I don't see a param in the API to set the DF bit. Will discuss with you!